### ISSUE TITLE
Port "Pangeo Forge Recipes" documentation section to use `beam-refactor` approach

### ISSUE THREAD

**Issue by derekocallaghan** [pangeo-forge/pangeo-forge-recipes#483]
The current documentation (#445) describes the Beam approach in the [Introduction Tutorial section](https://pangeo-forge--445.org.readthedocs.build/en/445/introduction_tutorial/index.html). The [Pangeo Forge Recipes section](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/index.html) will also need to be ported to replace e.g. `XarrayZarrRecipe` references with the recommended Beam equivalent.

I guess the final API is still a work in progress (e.g. see #473). If it's useful, I can add a new/update an existing [Recipe Tutorial](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/index.html) as this may help settle the approach recommended to recipe authors. We could the update the [Recipes User Guide](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/recipe_user_guide/index.html), [API Reference](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/api_reference.html) etc afterwards.

I have a local CCMP recipe that includes some custom processing (#453), which I can use to create a new recipe tutorial.

**The objective is to have an accurate (if reduced) version of the documentation by Feb 26th 2023.** The following sections will be ported:

- [x] [Recipes User Guide](https://pangeo-forge--499.org.readthedocs.build/en/499/pangeo_forge_recipes/recipe_user_guide/index.html) (#499)
- [Recipe Tutorials](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/index.html):
  - [x] [*Xarray-to-Zarr Sequential Recipe: NOAA OISST* notebook](https://pangeo-forge--487.org.readthedocs.build/en/487/pangeo_forge_recipes/tutorials/xarray_zarr/netcdf_zarr_sequential.html) (#487)
  - [x] [*NetCDF Zarr Sequential Recipe: CMIP6*](https://pangeo-forge--487.org.readthedocs.build/en/487/pangeo_forge_recipes/tutorials/xarray_zarr/cmip6-recipe.html) (#487)
  - [x] [*NetCDF Zarr Multi-Variable Sequential Recipe: NOAA World Ocean Atlas* notebook](https://pangeo-forge--487.org.readthedocs.build/en/487/pangeo_forge_recipes/tutorials/xarray_zarr/multi_variable_recipe.html) (#487)
  - [x] [*Complex NetCDF to Zarr Recipe: TerraClimate* ](https://pangeo-forge--490.org.readthedocs.build/en/490/pangeo_forge_recipes/tutorials/xarray_zarr/terraclimate.html) (#490)
  - [x] [*NARR: Subsetting and OPeNDAP*](https://pangeo-forge--500.org.readthedocs.build/en/500/pangeo_forge_recipes/tutorials/xarray_zarr/opendap_subset_recipe.html) (#500)
  - [x] [*HDF Reference Recipe for CMIP6*](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/hdf_reference/reference_cmip6.html) (https://github.com/pangeo-forge/pangeo-forge-recipes/pull/486)

- [ ] [API Reference](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/api_reference.html): check docstrings are accurate
- [x] [Development](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/development/index.html)
  * No Beam-related changes required.

**Comment by norlandrhagen** [comment]
Hey there @derekocallaghan, I would love to help out on porting over some of the [recipe tutorials.](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/index.html#xarrayzarrrecipe) Do you have any you prefer to work on or have already started on?

**Comment by derekocallaghan** [comment]
> 

Hi @norlandrhagen, I've just created #487 that contains an initial port of the *NetCDF Zarr Multi-Variable Sequential Recipe: NOAA World Ocean Atlas* notebook to use the `beam-refactor` approach.  Although it contains references to the [*Xarray-to-Zarr Sequential Recipe: NOAA OISST* notebook](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/xarray_zarr/netcdf_zarr_sequential.html), I haven't ported this yet as it contains functionality which may no longer be provided by the `beam-refactor` branch. I'm going to look at this next. 

In the meantime, the ported *NOAA World Ocean Atlas* notebook in #487 may be useful as an example for porting other tutorial notebooks. There are currently [five tutorial notebooks](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/index.html), so if you wanted to look at any of the following that'd be great:
* [*CMIP6*](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/xarray_zarr/cmip6-recipe.html)
* [*TerraClimate*](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/xarray_zarr/terraclimate.html)
* [*NARR: Subsetting and OPeNDAP*](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/xarray_zarr/opendap_subset_recipe.html)

It's possible these may take a number of iterations depending on API changes. I guess we can coordinate in this which notebooks are being worked on to avoid duplication.

**Comment by derekocallaghan** [comment]
Hi @rabernat, I'm looking into porting the [*Xarray-to-Zarr Sequential Recipe: NOAA OISST* notebook](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/xarray_zarr/netcdf_zarr_sequential.html) to use the Beam approach. Some of the steps are currently no longer fully relevant as they use functionality from [`pangeo_forge_recipes.recipes.xarray_zarr`](https://github.com/pangeo-forge/pangeo-forge-recipes/blob/45cc115f287ca7d33c33aaa343f21c862812c0bb/pangeo_forge_recipes/recipes/xarray_zarr.py) such as [`open_chunk()`](https://github.com/pangeo-forge/pangeo-forge-recipes/blob/45cc115f287ca7d33c33aaa343f21c862812c0bb/pangeo_forge_recipes/recipes/xarray_zarr.py#L397) e.g.:
* [Step 4: Play with the recipe](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/xarray_zarr/netcdf_zarr_sequential.html#step-4-play-with-the-recipe)
* [Step 5: Create storage targets](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/xarray_zarr/netcdf_zarr_sequential.html#step-5-create-storage-targets)
* [Step 6: Examine some chunks](https://pangeo-forge--445.org.readthedocs.build/en/445/pangeo_forge_recipes/tutorials/xarray_zarr/netcdf_zarr_sequential.html#step-6-examine-some-chunks)

Do we still want to have steps similar to these in the Beam version of the notebook, or replace them with steps that look at the output Zarr instead?

**Comment by cisaacstern** [comment]
Per conversation today during Coordination meeting, we've agreed to cut out the sections referenced above, which are no longer relevant in the context of beam-refactor.

**Comment by norlandrhagen** [comment]
@derekocallaghan here is the [TerraClimate recipe](https://github.com/pangeo-forge/pangeo-forge-recipes/pull/490).

**Comment by derekocallaghan** [comment]
An initial port of the [*Xarray-to-Zarr Sequential Recipe: NOAA OISST* notebook](https://pangeo-forge--487.org.readthedocs.build/en/487/pangeo_forge_recipes/tutorials/xarray_zarr/netcdf_zarr_sequential.html) has been created (#487).

**Comment by derekocallaghan** [comment]
**As there are a few sections to port, I've added a task list to this issue description to coordinate activity.**

**Comment by derekocallaghan** [comment]
I'm currently updating the remaining documentation to use the updated content from the tutorial notebooks (#487).

**Comment by cisaacstern** [comment]
Notes from Coordination Meeting today:

- Terraclimate blocked by open bug report (@cisaacstern link after meeting)
- Open separate issue for migrating tutorials from notebooks -> scripts (with CI testing)

Note: Latter is not a release blocker.

**Comment by derekocallaghan** [comment]
> Notes from Coordination Meeting today:
> 
>     * Terraclimate blocked by open bug report (@cisaacstern link after meeting)
> 
>     * Open separate issue for migrating tutorials from notebooks -> scripts (with CI testing)
> 
> 
> Note: Latter is not a release blocker.

Thanks @cisaacstern, apologies I couldn't make the last couple of meetings. I can help out with some of the notebook -> script migration.

**Comment by cisaacstern** [comment]
#490 is now in! I'll take one more spin through the docs this afternoon, and then merge `beam-refactor` 🚀 .

**Comment by cisaacstern** [comment]
Having skimmed the docs, my conclusion is that while there are certainly many opportunities for clarification and greater depth, there are no glaring inaccuracies, so I'm going to call this issue closed, and move on to releasing `beam-refactor`.
